### PR TITLE
Formatted Instant NFT Data

### PR DESCRIPTION
**Pull request by GhenadieVP**
Detect and display properly the Instant NFT Data type.

A given NFT Data field is consider an Instant type if:
- The kind is an I64.
- The type_name is Instant.
- The value is a valid i64.

Formatted the date with system's `format` to inherit all of the users settings regarding date formatting, there are no requirements to display the `Instant` NFT Data in a particular, fixed format.

IMG_0097 2

### CODE REVIEW

File: RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift [hunk 1]
```diff
@@ -279,9 +283,23 @@ private extension String {
...
+			} else {
+				.primitive(self)
+			}
+		}
+	}
 }
 
 private extension GatewayAPI.ProgrammaticScryptoSborValue {
+	enum TypeName: String {
+		case instant = "Instant"
```
Comment (kugel3) [inline RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift:300]: So "Instant" is the only typename we support? For non-instants, is `typeName` simply undefined?

**Review by kugel3 — APPROVED** [review]

File: RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift [hunk 2]
```diff
@@ -279,9 +283,23 @@ private extension String {
 			}
 		}
 	}
+
+	var asInstantDataField: ArbitraryDataFieldKind? {
+		nilIfEmpty.map {
+			if let timeInterval = Int64($0) {
```
Comment (Sajjon) [inline RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift:289]: Hmm this is an extension on `String`, so the i64 is sent as a String? That sounds strange and wrong?

Comment (GhenadieVP) [reply RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift:289]: yes, it is returned as String as per GW contract. i64 refers to the [scrypto sbor value kind](https://github.com/radixdlt/babylon-wallet-ios/blob/216b24d52913544c4d471f656224f301112230ef/RadixWallet/Clients/GatewayAPI/CodeGen/Input/gateway-api-schema.yml#L3454), and we then validate the payload as actually being the Int64

File: RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift (hunk 1 above)
Comment (GhenadieVP) [reply RadixWallet/Features/AssetsFeature/Components/NonFungibleAssetList/Components/Details/NonFungibleTokenDetails+View.swift:300]: `Instant` it is the only one we do identify and act upon currently. For non-instants we do ignore the typeName, even if it is defined.

**Review by CyonAlexRDX — APPROVED** [review]
LGTM

**Review by CyonAlexRDX — APPROVED** [review]
LGTM

**Pull request merged**